zmemory.h

```diff
@@ -0,0 +1,110 @@
+/* zmemory.h -- Private inline functions used internally in zlib-ng
+ * For conditions of distribution and use, see copyright notice in zlib.h
+ */
+
+#ifndef _ZMEMORY_H
+#define _ZMEMORY_H
+
+#if defined(__GNUC__) && (__GNUC__ >= 4)
+#  define HAVE_MAY_ALIAS
+#endif
+
+static inline uint16_t zng_memread_2(const void *ptr) {
+#if defined(UNALIGNED_OK)
+    return *(const uint16_t *)ptr;
+#elif defined(HAVE_MAY_ALIAS)
+    typedef struct { uint16_t val; } __attribute__ ((__packed__, __may_alias__)) unaligned_uint16_t;
+    return ((const unaligned_uint16_t *)ptr)->val;
+#else
+    uint16_t val;
+    memcpy(&val, ptr, sizeof(val));
+    return val;
+#endif
+}
+
+static inline uint32_t zng_memread_4(const void *ptr) {
+#if defined(UNALIGNED_OK)
+    return *(const uint32_t *)ptr;
```

Comment:
I just did a ton of work to avoid this kind of thing...
Compiler will convert `memcpy` to unaligned access if it is supported.

Reply:
These `UNALIGNED_OK` branches should be removed. So there is only `HAVE_MAY_ALIAS` and the `memcpy` path.

Reply:
I agree, `UNALIGNED_OK` branch should be removed.
We for sure don't want to default to that, if it was a new default-off setting like  `FORCE_UNALIGNED` or something, then there would at least be room for discussion.

The `HAVE_MAY_ALIAS` branch is interesting, I wonder what kind of performance that gets compared to memcpy on the various platforms/compilers. If we do want to use that, detection would preferably be in configure/CMake for cleaner and wider compiler support, or at the very least moved to `zbuild.h`.